### PR TITLE
Add option to add punctuation.

### LINKED ISSUE #60: Add support for punctuation
Monkeytyper has a mode that adds randomized punctuation to words in a test, such as a period/comma/colon/semicolon at the end of a word (periods making the next word capitalized), a dash between words, or surrounding the word with a delimiter, such as parentheses or quotes.

Would be nice to emulate this, to simulate english prose with all its punctuation without resorting to full book exerpts.

### PR DESCRIPTION
**Pull request by gabevenberg**
fixes #60, thanks for having a very clear, easy to work with codebase!

- [x] WordSelector implementation
- [x] command line options added
- [x] README documentation

### CODE REVIEW

File: src/textgen.rs [hunk 1]
```diff
@@ -248,3 +250,86 @@ impl<T: Seek + io::Read> WordSelector for RawWordSelector<T> {
...
+        let mut word = self.selector.new_word()?;
+
+        let will_punctuate = rng.gen_bool(self.punctuation_chance);
+        if will_punctuate || self.next_is_capital {
+            let mut chars: VecDeque<char> = word.chars().collect();
+            if self.next_is_capital {
+                // some unicode chars map to multiple chars when uppercased.
+                for c in chars.pop_front().expect("got empty word").to_uppercase() {
+                    chars.push_front(c)
+                }
```
Comment (Samyak2) [inline src/textgen.rs:312]: Would this add the characters in the reverse order?

For example, if `a` maps to `ABC`, will chars become:
```
['A']
['B', 'A']
['C', 'B', 'A']
```

**Review by Samyak2 — COMMENTED** [review]
This is great! Thank you for the contribution.

I just have one question ⬇️. Looks good to merge otherwise.

File: src/textgen.rs (hunk 1 above)
Comment (gabevenberg) [reply src/textgen.rs]: missed that, fixed.

**Review by Samyak2 — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**